packages/react-components/src/hooks/requestFunctions/useIoTEventsClient.ts

```diff
@@ -0,0 +1,33 @@
+import { IoTEvents, IoTEventsClient } from '@aws-sdk/client-iot-events';
+import { useMemo } from 'react';
+import { createDescribeAlarmModel } from './data/iotevents';
+
+/**
+ * Hook interface accepts any AWS SDK client type for IoT Events
+ */
+export interface UseIoTEventsClientOptions {
+  iotEventsClient?: IoTEventsClient | IoTEvents;
+}
+
+/**
+ *
+ * @param iotEventsClient is an AWS SDK client type for IoT Events
+ * @returns an IoTEvents type, promise pattern client
+ *
+ * @experimental Do not use in production.
+ */
+export function useIoTEventsClient({
+  iotEventsClient,
+}: UseIoTEventsClientOptions): IoTEvents {
+  const iotEvents = useMemo(() => {
+    if (iotEventsClient instanceof IoTEventsClient) {
+      return {
+        describeAlarmModel: createDescribeAlarmModel(iotEventsClient),
+      } as IoTEvents;
```

Comment:
instead can we use `satisfies` ?

Reply:
Not sure if `satisfies` applies here either. We want to explicitly type cast this because we use a different type for the apis. The type is a specific RequestFunction type that doesn't seem to cleanly satisfy the client type.